docs/decisions/0007-reducing-complexity.md

```diff
@@ -0,0 +1,39 @@
+# Reducing Complexity
+
+## Motivation
+
+High complexity is in general a concerning problem: It can lead to untested legacy code, a high amount of bugs, bad maintainability, slow development, and so on. These decisions are concrete changes we want to make to reduce complexity.
+
+## Decisions
+
+### Hooks
+
+Decision: Hooks always start with `use` (for example `useCounter`). This applies to custom hooks. A custom hook is any function that calls other hooks, especially state and lifecycle hooks like `useEffect` or `useState`. Functions that do not call other hooks never start with `use`.
```

Comment:
Is this a design decision or a coding convention? Sounds more like the latter.

Reply:
You could call it a coding convention, but it is more of a rule. We use React and React is very good in not being opiniated. However, React docs state custom hooks **MUST** start with "use" and this is not optional but mandatory. They give very good reasons for it (the link contains it all) and if we don't follow that we don't just create bad code design, but confuse other React developers who look at the code base.

Reply:
After some time to think: It doesn't matter much how you call it. What's relevant here is that you must be able to distinguish hooks from functions, and this is the only correct way to do it as decided by the React team. I can recommend reading more about it in the beta docs, it is enlightening.